packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -106,6 +125,20 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
     return null;
   }
 
+  double _getCollapsePadding(double t, _FlexibleSpaceBarSettings settings){
+    final double deltaExtent = settings.maxExtent - settings.minExtent;
+
+    switch(widget.collapseMode){
+      case CollapseMode.pin:
+        return -(settings.maxExtent - settings.currentExtent);
+      case CollapseMode.none:
+        return 0.0;
+      case CollapseMode.parallax:
+      default:
```

Comment:
remove the "default", since there's no other option here (that way the analyzer will tell us, whenever the enum has a new value added, that we need to update here)

Reply:
I made it throw an exception in the default ... since this method must return a double.
let me know if you have any other suggestion